File: 2-AndroidDesignPatterns/MVC/app/src/main/java/com/example/arch_lab1/model/local_source/ProductDAO.java
package com.example.arch_lab1.model.local_source;


import androidx.lifecycle.LiveData;
import androidx.room.Dao;
import androidx.room.Delete;
import androidx.room.Insert;
import androidx.room.OnConflictStrategy;
import androidx.room.Query;

import com.example.arch_lab1.model.pojo.Product;

import java.util.List;


@Dao
public interface ProductDAO {

    @Query("SELECT * FROM products_table")
    LiveData<List<Product>> getAllProducts();

    @Insert(onConflict = OnConflictStrategy.IGNORE)
    void insertProduct(Product product);

    @Delete
    void deleteProduct(Product product);


}

File: 4-MobileAppDevelopmentUsingKotlin/Day5/app/src/main/java/com/example/day5/ProductDao.kt
package com.example.day5

import androidx.room.Dao
import androidx.room.Insert
import androidx.room.OnConflictStrategy
import androidx.room.Query
import com.example.myapplication.Product

@Dao
interface ProductDao {
    @Insert(onConflict = OnConflictStrategy.REPLACE)
    suspend fun insertProducts(products: List<Product>)

    @Query("SELECT * FROM products")
    suspend fun getAllProducts(): List<Product>

    @Query("DELETE FROM products")
    suspend fun clearProducts()
}